operator/builtin/input/file/reader.go

```diff
@@ -29,15 +29,44 @@ import (
 	"github.com/open-telemetry/opentelemetry-log-collection/errors"
 )
 
+// File attributes contains information about file paths
+type fileAttributes struct {
+	Name         string
+	Path         string
+	ResolvedName string
+	ResolvedPath string
+}
+
+// resolveFileAttributes resolves file attributes
+// and sets it to empty string in case of error
+func resolveFileAttributes(path string) *fileAttributes {
+	resolved, err := filepath.EvalSymlinks(path)
+	if err != nil {
+		resolved = ""
+	}
+
+	abs, err := filepath.Abs(resolved)
+	if err != nil {
+		abs = ""
+	}
```

Comment:
Best I can tell, both `filepath.EvalSymlinks` and `filepath.Abs` will only ever return `"", err`, or `"not empty", nil`, so these error checks aren't really doing anything.

Moreover, I think we should return the errors and log them in the callers. In both cases (`newReader` and `NewReader`), we have a handle to the operator, which has a logger.

Reply:
Not sure how to properly implement it. For now all errors will result in no data ingestion:

https://github.com/open-telemetry/opentelemetry-log-collection/blob/main/operator/builtin/input/file/file.go#L283-L286

Should I exclude filepath related errors from this condition and just print logs for them?

Reply:
I see your point. Probably then it makes sense to just make this a receiver on the operator:

```go
func (f *InputOperator) resolveFileAttributes(path string) *fileAttributes {
	resolved, err := filepath.EvalSymlinks(path)
	if err != nil {
		f.Error(err)
	}

	abs, err := filepath.Abs(resolved)
	if err != nil {
		f.Error(err)
	}

	return ...
}
```